java-checks-test-sources/src/main/files/non-compiling/checks/unused/UnusedMethodParameterCheck.java

```diff
@@ -1,7 +1,39 @@
 class UnusedMethodParameterCheck {
 
-  public void unknownAnnotatedParameter(@Unknown Object event, int arg2) { // Compliant, unknown annotation on parameter
+  public void unknownAnnotatedParameter(@Unknown Object agr1, int arg2) { // Compliant, unknown annotation on parameter
     System.out.println(arg2);
   }
 
+  abstract class KnownParent {
+    public abstract void method(Object arg1, int arg2);
+    public abstract void methodWithUnknowParameterType(Unknown arg1, int arg2);
+  }
+
+  class ChildWithKnownHierarchy extends KnownParent {
+
+    public void nonOverringMethod(Object arg1, int arg2) { // Noncompliant
+      System.out.println(arg2);
+    }
+
+    public void method(Object arg1, int arg2) { // Compliant, overriding
+      System.out.println(arg2);
+    }
+
+    public void methodWithUnknowParameterType(Unknown arg1, int arg2) { // Compliant, overriding
+      System.out.println(arg2);
+    }
+  }
```

Comment:
An override method should have the exact same argument types, so those two test cases should be `Noncompliant`.
And because of the limitation of our semantics, the first case is a false-negative.